hydra-node/test/Hydra/Chain/Direct/StateSpec.hs

```diff
@@ -277,6 +300,17 @@ spec = parallel $ do
     it "can close & fanout every collected head" $ do
       prop_canCloseFanoutEveryCollect
 
+genAdaOnlyUTxOOnMainnetWithAmountBiggerThanOutLimit :: Gen UTxO
+genAdaOnlyUTxOOnMainnetWithAmountBiggerThanOutLimit = do
+  adaAmount <- (+ maxMainnetLovelace) . getPositive <$> arbitrary
+  utxo <- genUTxO1 genTxOut
+  let utxoPairs =
+        ( \(a, b) ->
+            (a, modifyTxOutValue (const $ lovelaceToValue (Lovelace adaAmount)) b)
+        )
+          <$> UTxO.pairs utxo
+  pure . UTxO.UTxO $ Map.fromList utxoPairs
```

Comment:
Could: Use the `Functor` instance of `UTxO` or apply it to the `genTxOut` directly.

This will make this very short and I think it could be inlined.

```suggestion
  genUTxO1 $ do
    lovelace <- lovelaceToValue . Lovelace . (+ maxMainnetLovelace) . getPositive <$> arbitrary
    modifyTxOutValue (const lovelace) <$> genTxOut
```